### ISSUE TITLE
New Installation Issues

### ISSUE THREAD

**Issue by mygeek911** [nextcloud/vm#2448]
### Steps To Reproduce

1. Install using script as documented [here](https://docs.hanssonit.se/s/bj0vl1ihv0jgrmfm08j0/build-your-own/d/bj0vl4ahv0jgrmfm0950/nextcloud-vm) with new barebones installation of Ubuntu server with second hard drive on HyperV server
2. Open Web browser and go to IP address or domain noted at end of installation
3. Click on link found under "Access Nextcloud" section with the following address: [https://172.16.ip.address (HTTPS)](#)

![front-page](https://user-images.githubusercontent.com/8011893/223252527-91456339-a588-4239-a764-b1d9b5f8dcdf.png)

### Expected Result

I would expect the login page to appear under the root IP address or domain.

### Actual Result

The correct login URL appears ( [https://172.16.ip.address/login](#) ), but the page shows a 404 error.

![not-found](https://user-images.githubusercontent.com/8011893/223252902-f81e4e5d-0c0b-418a-805e-a01e4cd0a811.png)

### Additional Context

The only way to access the site is within the **nextcloud** folder ( [https://172.16.ip.address/nextcloud](#) ). I tried changing the `htaccess.ReWriteBase` value in the `/var/www/nextcloud/config/config.php` file to include nextcloud at the end, but it doesn't recognize the change, even after restarting apache2 or rebooting the server. I still have to go to the original link above.

As a test, I tried installing twice: once with the default recommendations and then again with no additional apps or anything else. Both times I encountered the same issue.

I'd really like Nextcloud to run from the root domain or IP address if possible.

### Build Version

25.0.4 - 25.0.4.1

### Environment

By using the scripts

### Environment Details

Server configuration detail

**Operating system:** Linux 5.15.0-1034-azure # 41-Ubuntu SMP Fri Feb 10 19:59:45 UTC 2023 x86_64

**Webserver:** Apache (fpm-fcgi)

**Database:** pgsql PostgreSQL 14.7 (Ubuntu 14.7-0ubuntu0.22.04.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0, 64-bit

**PHP version:** 8.1.2-1ubuntu2.11

**Modules loaded:** Core, date, libxml, openssl, pcre, zlib, filter, hash, json, Reflection, SPL, session, standard, sodium, cgi-fcgi, PDO, xml, bcmath, bz2, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imap, intl, ldap, exif, pdo_pgsql, pgsql, Phar, posix, readline, redis, shmop, SimpleXML, smbclient, soap, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, libsmbclient, Zend OPcache

**Nextcloud version:** 25.0.4 - 25.0.4.1

**Comment by mygeek911** [comment]
This was resolved when I ran the `/var/scripts/menu.sh` script again. I don't know which of the following modifications I made caused it to suddenly work:

- CookieLifetime
- Share-folder
- Check 0-byte files
- Update mimetype list
- Enable logrotate
- Activate TLS (which actually failed)

I've got a feeling it was something that modified the `/var/www/nextcloud/config/config.php` file.

**Comment by mygeek911** [comment]
So what actually fixed it was running the `Activate TLS` script from `/var/scripts/menu.sh`. Since I'm behind a proxy and I hadn't got the .well-known folder working, the first time I ran it the script failed (accidentally because I left off a character in the TXT record for DNS verification), but Nextcloud was able to log in from the root folder.

Some time later, I ran the TLS script again and properly populated the TXT record, and received a cert for the Nextcloud server itself. But...

After a few minutes the root website as connected to by domain name decided to forward automatically to the _private_ IP address of the Nextcloud server. Well, that doesn't work so well when the hosting server is hosted elsewhere.

Long story short, I'm behind Cloudflare so I can use their SSL cert for the entire chain. I had just hoped to get a cert going on Nextcloud so that communications would be encrypted to and from the proxy server.

I've reinstalled Nextcloud and ran the TLS script, but had it fail again. The connection is working, but it appears there are a couple of bugs in the process.

**Comment by enoch85** [comment]
Thanks for your report and investigation!

> Since I'm behind a proxy

I think the issue is that one. I can't reproduce this if not behind a proxy, even with all settings enabled.

For TLS you coould use deSEC. It won't care which port you're using and if you have ports open or not. You can then put a CNAME to your real domain from the dedyn domain.

Hope it helps!

**Comment by mygeek911** [comment]
I didn't explain that as well as I had hoped. The proxy wasn't used when I was on the local network.

Try going through the initial setup, but not requesting a TLS cert. Once setup is finished and just on the local network, not behind a proxy, clicking the root IP address link to log in to the site should get you that 404 error. It is only _after_ you run the TLS script (whether it fails or not), then the root IP address will work.

**Comment by enoch85** [comment]
OK, thanks for clarifying. Will try to reproduce.

**Comment by enoch85** [comment]
[Screencast from 2023-03-07 13:49:40.webm](https://user-images.githubusercontent.com/4511254/223426956-7708b15c-3344-4b74-a2ee-abc8c0d86aa1.webm)

Nope, not reproducible on the latest VM. This was made on Firefox.

**Comment by mygeek911** [comment]
Interesting. I must be the exception. Ha ha. There must be one non-standard setting in my case, and since I can correct for it, then it isn't as terrible as it seems.

I appreciate you looking into it. If you would like, I can spend time later on this evening on a completely different HyperV server to see if I can duplicate it there.

**Comment by mygeek911** [comment]
I apologize for the lateness in getting this done. The install went fine without a hitch on a different server. There must just be something different on the original one. Seems this issue is resolved!

**Comment by enoch85** [comment]
🥷👍🏼
